permissions_validators/src/public_blockchain/burn.rs

```diff
@@ -1,17 +1,26 @@
 //! Module with permission for burning
 
-use std::str::FromStr as _;
+use iroha_data_model::asset::DefinitionId;
 
 use super::*;
 
-/// Can burn asset with the corresponding asset definition.
-#[allow(clippy::expect_used)]
-pub static CAN_BURN_ASSET_WITH_DEFINITION: Lazy<Name> =
-    Lazy::new(|| Name::from_str("can_burn_asset_with_definition").expect("normal name"));
-#[allow(clippy::expect_used)]
-/// Can burn user's assets permission token name.
-pub static CAN_BURN_USER_ASSETS_TOKEN: Lazy<Name> =
-    Lazy::new(|| Name::from_str("can_burn_user_assets").expect("normal name"));
+declare_token!(
+    /// Can burn asset with the corresponding asset definition.
+    CanBurnAssetWithDefinition {
+        /// Asset definition id.
+        asset_definition_id ("asset_definition_id"): DefinitionId,
+    },
+    "can_burn_asset_with_definition"
+);
```

Comment:
I chose to leave it as macro_rules because I don't have a lot of experience writing procedural macros and this one is straightforward enough to be understandable in spite of it's length IMO.

Reply:
Let's create a ticket for it in chores.